src/rgw/rgw_sal_motr.cc

```diff
@@ -1825,10 +1825,13 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
   if (rc < 0)
     goto out;
 
-  actual = bs;
-  for (; off < end; off += actual) {
-    if (end - off < bs)
-        actual = end - off;
+  left = end - off;
+  for (; left > 0; off += actual) {
+    if (left < bs)
+      bs = this->get_optimal_bs(left);
+    actual = bs;
+    if (left < bs)
```

Comment:
Add comment stating that bs may remain same, and hence this additional check

Reply:
Sorry, I cannot get what you mean. `bs` can change - and that's why the additional check is needed.

Reply:
This check is not for "bs value might not change" . This check is for if the new bs value is still more than `left` , in that case `actual` should be equal to the `left`. 

This comment is not appropriate and should be removed.